packages/block-library/src/post-excerpt/edit.js

```diff
@@ -16,23 +16,27 @@ import {
 	Warning,
 	useBlockProps,
 } from '@wordpress/block-editor';
-import { PanelBody, ToggleControl } from '@wordpress/components';
-import { __ } from '@wordpress/i18n';
+import { PanelBody, ToggleControl, RangeControl } from '@wordpress/components';
+import { sprintf, __, _x } from '@wordpress/i18n';
+import { count as wordCount } from '@wordpress/wordcount';
+import { speak } from '@wordpress/a11y';
+import { useDebounce } from '@wordpress/compose';
 
 /**
  * Internal dependencies
  */
 import { useCanEditEntity } from '../utils/hooks';
 
 export default function PostExcerptEditor( {
-	attributes: { textAlign, moreText, showMoreOnNewLine },
+	attributes: { textAlign, moreText, showMoreOnNewLine, excerptLength },
 	setAttributes,
 	isSelected,
 	context: { postId, postType, queryId },
 } ) {
 	const isDescendentOfQueryLoop = Number.isFinite( queryId );
 	const userCanEdit = useCanEditEntity( 'postType', postType, postId );
 	const isEditable = userCanEdit && ! isDescendentOfQueryLoop;
+	const debouncedSpeak = useDebounce( speak, 3000 );
```

Comment:
I only tested this delay by my own ear using VoiceOver, so it might be too long for users who are more used to listening to the screen reader?

Reply:
cc @afercia

Reply:
Great to see `speak` for this use case 🎉 A few considerations:
- Three seconds seem a bit too much to me. I'd go with 500ms: looking at other occurrences of a debounced `speak` call, that seems the de facto standard. If I remember correctly, 500ms is the same delay used in core in a few places. That said, ideally this value should be adjustable by users. Not all users type at the same speed. Useers who type slowly would benefit from an adjustable delay but this is out of the scope of this PR.
- To my understanding, right now the spoken message occurs at every React render, even when the actual count hasn't changed. That is, the _visual_ counter doesn't change but a new message is announced anyways at any ley press (after the delay). Not sure the mismatch between visual and audible is ideal. Repeating the message each time may be a bit noisy. I'd try to send a message only when `currentWordCount` or `excerptLength` actually change. Maybe `useEffect` may come in handy for that.
- I'd consider to improve the quality of the audible message. As a blind user, hearing messages like:
```
8  10
9  10
10  10
minus 1 
minus 2
minus 3
etc.
```
wouldn't help me that much as the context isn't clear. Ideally, the spoken messages should use specifically crafted, more meaningful, sentences.